readthedocs/oauth/services/base.py

```diff
@@ -200,25 +205,51 @@ def sync(self):
         # Delete RemoteRepository where the user doesn't have access anymore
         # (skip RemoteRepository tied to a Project on this user)
         all_remote_repositories = remote_repositories + remote_repositories_organizations
-        repository_full_names = [r.full_name for r in all_remote_repositories if r is not None]
+        repository_remote_ids = [r.remote_id for r in all_remote_repositories if r is not None]
         (
-            self.user.oauth_repositories
+            self.user.remote_repository_relations
             .exclude(
-                Q(full_name__in=repository_full_names) | Q(project__isnull=False)
+                remote_repository__remote_id__in=repository_remote_ids,
+                remote_repository__vcs_provider=self.vcs_provider_slug
             )
             .filter(account=self.account)
             .delete()
         )
 
         # Delete RemoteOrganization where the user doesn't have access anymore
-        organization_slugs = [o.slug for o in remote_organizations if o is not None]
+        organization_remote_ids = [o.remote_id for o in remote_organizations if o is not None]
         (
-            self.user.oauth_organizations
-            .exclude(slug__in=organization_slugs)
+            self.user.remote_organization_relations
+            .exclude(
+                remote_organization__remote_id__in=organization_remote_ids,
+                remote_organization__vcs_provider=self.vcs_provider_slug
+            )
             .filter(account=self.account)
             .delete()
         )
 
+    def get_remote_repository_relation(self, repo):
```

Comment:
Should this be on the model, instead of a util here?

Reply:
We are only using this here and we already have access to the account and user object here as well, that's why felt like it should be here so that we don't need to pass these to other places, but we can probably move it to the models.

Reply:
I also prefer to have the code contained here, in particular if it's not required outside, and also to avoid keep adding lot of model methods if they are not strictly required there.

Reply:
Don't worry too much. I'm fine having it on this model as well. I thought it was ending on the `Project` model, but I was wrong.